### PR TITLE
Add bom to project

### LINKED ISSUE #466: Add BOM
### Issue description

We should add a BOM to this module.

### PR DESCRIPTION
**Pull request by timyates**
Fixes #466

### CODE REVIEW

**Comment by timyates** [comment]
I have no idea why we get 0% coverage

The tests did run, so I'd expect at least _some_ coverage

https://ge.micronaut.io/s/jjnsl4lz5udak/tests/overview

Maybe it's because `master` is yet to be scanned or something

File: grpc-server-runtime/build.gradle [hunk 1]
```diff
@@ -50,9 +42,9 @@ sourceSets {
 }
 
 protobuf {
-    protoc { artifact = "com.google.protobuf:protoc:$protobufJavaVersion" }
+    protoc { artifact = libs.protoc.asProvider().get() }
     plugins {
-        grpc { artifact = "io.grpc:protoc-gen-grpc-java:$grpcVersion" }
+        grpc { artifact = libs.protoc.grpc.get() }
     }
```
Comment (timyates) [inline grpc-server-runtime/build.gradle:48]: @melix, is this correct?  It works, but I'm not 100% sure it's sensible 😀

**Review by melix — APPROVED** [review]

File: gradle.properties [hunk 2]
```diff
@@ -3,13 +3,12 @@ projectGroup=io.micronaut.grpc
 micronautDocsVersion=2.0.0
 micronautVersion=3.1.3
 micronautTestVersion=3.0.3
+
 micronautGradlePluginVersion=2.0.8
```
Comment (sdelamo) [inline gradle.properties:7]: what is this for? `micronautGradlePluginVersion`?

File: gradle.properties [hunk 3]
```diff
@@ -21,3 +20,5 @@ githubBranch=master
 githubCoreBranch=3.2.x
```
Comment (sdelamo) [inline gradle.properties:20]: set  `githubCoreBranch` to `3.5.x`

File: src/main/docs/guide/server.adoc [hunk 4]
```diff
@@ -154,7 +153,7 @@ The above example uses the `@GrpcChannel` annotation to inject a gRPC `ManagedCh
 
 Now that you have a test client, writing the test becomes trivial:
 
-snippet::helloworld.GreetingEndpointTest[tags="imports,test", project-base="examples/hello-world"]
+snippet::helloworld.GreetingEndpointTest[tags="imports,test", project-base="test-suite"]
```
Comment (sdelamo) [inline src/main/docs/guide/server.adoc:156]: if you use `test-suite`, you don't need to specify a `project-base` for the `snippet` macro.

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by sdelamo** [comment]
@timyates Thanks a lot of this PR!

File: gradle.properties (hunk 2 above)
Comment (timyates) [reply gradle.properties:7]: They are used here:

https://github.com/micronaut-projects/micronaut-grpc/blob/8b5f55876bcb12ea50e2c9ea53ef2c73b616cd25/settings.gradle#L3-L4

We can't use catalogs for plugin versions, and I decided to leave these in the properties file as I guessed that would be the first place people would look for them

File: src/main/docs/guide/server.adoc (hunk 4 above)
Comment (timyates) [reply src/main/docs/guide/server.adoc:156]: Done! Also updated the version to 3.2.0-SNAPSHOT

File: gradle.properties (hunk 3 above)
Comment (timyates) [reply gradle.properties]: Done! Also updated the version to 3.2.0-SNAPSHOT

File: src/main/docs/guide/server.adoc (hunk 4 above)
Comment (timyates) [reply src/main/docs/guide/server.adoc]: Done 👍

**Comment by timyates** [comment]
> @timyates Thanks a lot of this PR!

A pleasure as always 😊

**Pull request merged**